### ISSUE TITLE
Weird vibrations on iOS distro

### ISSUE THREAD

**Issue by C29H25N3O5** [26F-Studio/Techmino#256]
When adjusting the Vibration slider in Settings, the strength of vibration cannot be specified on iOS devices.
Even when the strength was set to 10, the device still produces the same vibration strength as if the strength is 1.
It acts more like switch than a slider.

Also, it is suggested that the duration of the vibration could be shortened. Currently, the duration is so long that when multiple press was conducted in a short time, the device only produces one vibration.

I don't know whether this issue occurs on Android devices but it is a major problem on iPhone.

**Comment by Trebor-Huang** [comment]
This is impossible to implement without changing the source code of love and introducing undocumented API's of Apple, breaking support on older iOS versions.

**Comment by Trebor-Huang** [comment]
This is impossible to implement without changing the source code of love and introducing undocumented API's of Apple, breaking support on older iOS versions.

**Comment by Trebor-Huang** [comment]
> I'm pretty sure it's possible to check the OS version on LOVE, and you can adjust it to that

No, the change of code is done outside of this repository, and outside of LOVE runtime. We might need an iOS developer to help with that. Lua code won't do here.

**Comment by C29H25N3O5** [comment]
> > I'm pretty sure it's possible to check the OS version on LOVE, and you can adjust it to that
> 
> No, the change of code is done outside of this repository, and outside of LOVE runtime. We might need an iOS developer to help with that. Lua code won't do here.

Alright then. If that is the case, is it possible to just remove the vibration slider on iOS (before an LÖVE update to solve this problem)?

**Comment by Trebor-Huang** [comment]
Good news! We're working to do exactly the change I've described. But removing that slider before the work's done seems reasonable.

**Comment by haolloyin** [comment]
`love2d` uses a very old API `AudioServicesPlaySystemSound(kSystemSoundID_Vibrate);` which can only vibrate for about 0.5 seconds. See https://love2d.org/wiki/love.system.vibrate

![image](https://user-images.githubusercontent.com/1009628/133228118-43ef9ba0-c781-4137-8f3f-a4440d0c25d3.png)


The implementation is here: https://github.com/love2d/love/blob/master/src/common/ios.mm#L394-L400

The declaration is here: https://github.com/love2d/love/blob/master/src/common/ios.h#L67-L69 and it says `Causes devices with vibration support to vibrate for about 0.5 seconds.` is actually incorrect, because since iOS 10, there is a new API `UIImpactFeedbackGenerator` that can generate very light vibrations (requires iOS 10+).

The iOS development documentation is available at https://developer.apple.com/documentation/uikit/uiimpactfeedbackgenerator?language=objc


Only 7% of iOS users worldwide are now using iOS 12 or lower, Apple's official source https://developer.apple.com/support/app-store/

![image](https://user-images.githubusercontent.com/1009628/133228002-7105aa7e-a15b-4470-a6b1-447f81ff8d73.png)


So, if Techmino's iOS developers manually change `love2d`'s `src/common/ios.mm#L398` to the following when packaging it would solve the problem.

```objc
if (@available(iOS 10.0, *)) {
    UIImpactFeedbackGenerator *feedBackGenertor = [[UIImpactFeedbackGenerator alloc] initWithStyle:UIImpactFeedbackStyleMedium];
    [feedBackGenertor impactOccurred];
} else {
    AudioServicesPlaySystemSound(kSystemSoundID_Vibrate);
}

```

**Comment by ParticleG** [comment]
Fixed On latest [PR#3](https://github.com/26F-Studio/Techmino-iOS/pull/3) of `Techmino-iOS` repo

**Comment by ParticleG** [comment]
Okay, now we need to add `iOS 13.0+` API for even finer controls[impactOccurredWithIntensity](https://developer.apple.com/documentation/uikit/uiimpactfeedbackgenerator/3183920-impactoccurredwithintensity?language=objc).
Also need to change: 
![image](https://user-images.githubusercontent.com/38580644/220542352-247a2cff-3694-4e7f-a726-47e11e4aaca7.png)
